scripts/release-docs.ts

```diff
@@ -0,0 +1,30 @@
+import { execSync } from 'child_process';
+import { gt, major, maxSatisfying } from 'semver';
+
+// The GITHUB_REF_NAME is a full version (i.e. 17.3.2). The branchName will strip the patch version number.
+// We will publish docs to the website branch based on the current tag (i.e. website-17)
+const currentVersion = process.env.GITHUB_REF_NAME || '';
+console.log(`Comparing ${currentVersion} to npm versions`);
+
+const majorVersion = major(currentVersion);
+const releasedVersions: string[] = JSON.parse(
+  execSync(`npm show nx@^${majorVersion} version --json`).toString()
+);
+
+const latestVersion = maxSatisfying(releasedVersions, `^${majorVersion}`);
+
+console.log(`Found npm versions:\n${releasedVersions.join('\n')}`);
+
+// Publish if the current version is greater than the latest released version
+
+const branchName = `website-${majorVersion}`;
+if (currentVersion && latestVersion && gt(currentVersion, latestVersion)) {
```

Comment:
This happens after the release...

* `currentVersion` will be... the one which was just published
* `latestVersion` will be.... maybe the one which was just published. It won't be higher though. So I don't think this condition will pass...

I think it needs to be equal to the max satisfying...

Does that make sense or am I missing something?